src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -292,6 +644,58 @@ impl MetaTxn for EtcdTxn {
         self.buffer.insert(key, None);
     }
 
+    /// Try to campaign the master with simple txn
+    /// Old(reference to etcd client v3):
+    /// Create a key with prefix, and compare key (sorted by revision) with the key created by the same session.
+    /// If current key is the smallest, then the session is the master, and campaign success.
+    ///
+    /// New(For datenlord cache scenario):
+    /// A small batch of nodes in cluster, we just need a simple txn to get the master key.
+    /// The thundering herd is not the main problem, which will cause a bunch of etcd raft logs.
+    async fn campaign(&self, key: &KeyType, val: &ValueType, lease_id: i64) -> bool {
+        let mut client = self.client.clone();
+
+        // Try to get the key, if key is existed
+        // We need to return the data from etcd,
+        // If the key is not existed, we need to set the key.
+        let txn = Txn::new()
+            .when(vec![Compare::create_revision(
+                key.to_string_key(),
+                CompareOp::Equal,
+                0,
+            )])
+            .and_then(vec![TxnOp::put(
+                key.to_string_key(),
+                serde_json::to_string(val).unwrap_or_else(|val| {
+                    panic!("failed to serialize value to json, value = {val:?}");
+                }),
+                Some(PutOptions::new().with_lease(lease_id)),
+            )])
+            .or_else(vec![TxnOp::get(
+                key.to_string_key(),
+                Some(GetOptions::new().with_serializable()),
+            )]);
+
+        let res = match client.txn(txn).await {
+            Ok(resp) => {
+                // Check the txn branch is `then` or `else`
+                if resp.succeeded() {
```

Comment:
Should deal with resp type and value,    because put resp means campaign success, and get resp means the key exists,  at lease return the etcd key value so that does not require a second query.

Reply:
Ok, I try to return both status and key in this function.